src/client/dfuse/dfuse_core.c

```diff
@@ -270,18 +270,6 @@ _ph_free(struct dfuse_pool *dfp)
 
 	if (daos_handle_is_valid(dfp->dfp_poh)) {
 		rc = daos_pool_disconnect(dfp->dfp_poh, NULL);
-		/* Hook for fault injection testing, if the disconnect fails with out of memory
-		 * then simply try it again, only what might have happened is that the first
-		 * call might have disconnected, but then failed to notify about the disconnect,
-		 * in which case the subsequent call will return -DER_NO_HDL, if that's the case
-		 * then this is expected, if odd, behavior so silence that case and just return
-		 * success.
-		 */
-		if (rc == -DER_NOMEM) {
```

Comment:
wouldn't this change break FI testing?

Reply:
No, because in this case the underlying behavior issue is fixed.

Reply:
As Ashley pointed out below, you are right after all. I thought its intent was only to fix this specific odd case with the agent notification.